ksqldb-functional-tests/src/test/resources/query-validation-tests/suppress.json

```diff
@@ -15,10 +91,239 @@
+      "statements": [
+        "CREATE STREAM INPUT (ID STRING KEY, COL1 STRING) WITH (kafka_topic='input_topic',value_format='JSON');",
+        "CREATE TABLE OUTPUT AS SELECT ID, COUNT(*) as COUNT FROM INPUT WINDOW HOPPING (SIZE 5 MILLISECONDS,ADVANCE BY 2 MILLISECONDS, GRACE PERIOD 1 MILLISECONDS) GROUP BY ID EMIT FINAL;"
+      ],
+      "inputs": [
+        {"topic": "input_topic", "key": "k1", "value": {"COL1": "v1"},"timestamp": 0},
+        {"topic": "input_topic", "key": "k1", "value": {"COL1": "v1"},"timestamp": 1},
+        {"topic": "input_topic", "key": "k1", "value": {"COL1": "v1"},"timestamp": 2},
+        {"topic": "input_topic", "key": "k1", "value": {"COL1": "v1"},"timestamp": 1},
+        {"topic": "input_topic", "key": "k1", "value": {"COL1": "v1"},"timestamp": 6},
+        {"topic": "input_topic", "key": "k1", "value": {"COL1": "v1"},"timestamp": 5},
+        {"topic": "input_topic", "key": "k1", "value": {"COL1": "v1"},"timestamp": 10}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": "k1", "value": {"COUNT": 4},"window": {"start": 0, "end": 5, "type": "time"}, "timestamp": 2},
+        {"topic": "OUTPUT", "key": "k1", "value": {"COUNT": 3},"window": {"start": 2, "end": 7, "type": "time"},"timestamp": 6},
+        {"topic": "OUTPUT", "key": "k1", "value": {"COUNT": 2},"window": {"start": 4, "end": 9, "type": "time"},"timestamp": 6}
+      ]
+    },
+    {
+      "name": "should suppress multiple keys",
+      "statements": [
+        "CREATE STREAM INPUT (ID STRING KEY, COL1 STRING) WITH (kafka_topic='input_topic',value_format='JSON');",
+        "CREATE TABLE OUTPUT AS SELECT ID, COUNT(*) as COUNT FROM INPUT WINDOW TUMBLING (SIZE 2 MILLISECONDS, GRACE PERIOD 1 MILLISECONDS) GROUP BY ID EMIT FINAL;"
+      ],
+      "inputs": [
+        {"topic": "input_topic", "key": "k1", "value": {"COL1": "v1"},"timestamp": 0},
+        {"topic": "input_topic", "key": "k2", "value": {"COL1": "v1"},"timestamp": 1},
+        {"topic": "input_topic", "key": "k1", "value": {"COL1": "v1"},"timestamp": 2},
+        {"topic": "input_topic", "key": "k1", "value": {"COL1": "v1"},"timestamp": 1},
+        {"topic": "input_topic", "key": "k2", "value": {"COL1": "v1"},"timestamp": 0},
+        {"topic": "input_topic", "key": "k1", "value": {"COL1": "v1"},"timestamp": 5},
+        {"topic": "input_topic", "key": "k1", "value": {"COL1": "v1"},"timestamp": 0},
+        {"topic": "input_topic", "key": "k2", "value": {"COL1": "v1"},"timestamp": 2}
+
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": "k1", "value": {"COUNT": 2},"window": {"start": 0, "end": 2, "type": "time"},"timestamp": 1},
+        {"topic": "OUTPUT", "key": "k2", "value": {"COUNT": 2},"window": {"start": 0, "end": 2, "type": "time"},"timestamp": 1},
+        {"topic": "OUTPUT", "key": "k1", "value": {"COUNT": 1},"window": {"start": 2, "end": 4, "type": "time"},"timestamp": 2}
+      ]
+    },
+    {
+      "name": "should suppress after join",
```

Comment:
What do you think about also adding a test for join after suppression? I don't know if you're allowed to join windowed streams, which may disqualify this idea.

Reply:
I am not sure either but maybe @agavra knows? Though my understanding was that suppression here will return a table with windowed keys, and the only windowed joins that ksqlDB supports is that between two streams? I may be misunderstanding the joins possible here though

Reply:
I don't think we allow windowed sources to any operation at the moment, so probably no need to test that here